### PR TITLE
110 - Fetch Token Mint Info

### PR DESCRIPTION
**Pull request by stevenhansel**
This is the initial pull request for fetch token mint info. Probably there needs a discussion on how to handle if the token info is not found at all (from both Metaplex and Solana). Right now, it throws an application error (because it tries to access a null object). Also, other than displaying the symbol, do you want to display something else? e.g. the token logo

### CODE REVIEW

File: src/api/otc-state/fetchTokenInfo.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
+import { findMetadataPda } from '@metaplex-foundation/js';
```
Comment (thegreat-vanderlinde) [inline src/api/otc-state/fetchTokenInfo.ts:1]: can we move this utility file in `api/tokens/..` as it's something not strictly related to the otc state

File: src/api/otc-state/fetchOtcState.ts [hunk 2]
```diff
@@ -25,6 +26,8 @@ export const fetchOtcState = async (provider: AnchorProvider, otcStateAddress: P
 	const res = new OtcState();
 	res.publickey = otcStateAddress;
 	res.reserveMintInfo = await getMint(provider.connection, trancheConfigAccountInfo.reserveMint);
+	res.reserveTokenInfo = await fetchTokenInfo(provider.connection, trancheConfigAccountInfo.reserveMint);
```
Comment (thegreat-vanderlinde) [inline src/api/otc-state/fetchOtcState.ts:29]: here we can add this check:
```ts
if(trancheConfigAccountInfo.reserveMint.toBase58() === "7XSvJnS19TodrQJSbjUR6tEGwmYyL1i9FX7Z5ZQHc53W")
   res.reserveTokenInfo = await fetchTokenInfo(provider.connection, new PublicKey("EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v"));
else
   res.reserveTokenInfo = await fetchTokenInfo(provider.connection, trancheConfigAccountInfo.reserveMint);
```
this way we'll simulate the behavior of `7XSvJnS19TodrQJSbjUR6tEGwmYyL1i9FX7Z5ZQHc53W` (our devnet dummy token) as it's [USDC](https://explorer.solana.com/address/EPjFWdd5AufqSSqeM2qN1xzybapC8G4wEGGkZwyTDt1v)

**Review by thegreat-vanderlinde — CHANGES_REQUESTED** [review]

**Comment by thegreat-vanderlinde** [comment]
@stevenhansel PR looks fine, can you just take care of conflicts and the comments?
For now in the UI is enough what you did, just the token symbol (USDC)

**Comment by stevenhansel** [comment]
> @stevenhansel PR looks fine, can you just take care of conflicts and the comments? For now in the UI is enough what you did, just the token symbol (USDC)

Thank you for the comments and also for all the guidance in this issue, I just pushed the commit to handle the conflicts and comments. Please kindly recheck it, thanks!

**Review by thegreat-vanderlinde — APPROVED** [review]
lgtm

**Pull request merged**